scripts/release.sh

```diff
@@ -0,0 +1,568 @@
+            -p|--prerelease)
+                PRERELEASE=true
+                shift
+                ;;
+            -d|--dry-run)
+                DRY_RUN=true
+                shift
+                ;;
+            -h|--help)
+                usage
+                exit 0
+                ;;
+            *)
+                log_error "Unknown option: $1"
+                usage
+                exit 1
+                ;;
+        esac
+    done
+
+    # Validate arguments
+    if [[ -z "$VERSION" && -z "$VERSION_TYPE" && "$PRERELEASE" != true ]]; then
+        log_error "Either --version, --type, or --prerelease (alone) must be specified"
+        usage
+        exit 1
+    fi
+
+    if [[ -n "$VERSION" && -n "$VERSION_TYPE" ]]; then
+        log_error "Cannot specify both --version and --type"
+        usage
+        exit 1
+    fi
+
+    if [[ -n "$VERSION_TYPE" && ! "$VERSION_TYPE" =~ ^(major|minor|patch)$ ]]; then
+        log_error "Version type must be one of: major, minor, patch"
+        exit 1
+    fi
+
+    # If only --prerelease is specified, we'll increment prerelease number only
+    if [[ -z "$VERSION" && -z "$VERSION_TYPE" && "$PRERELEASE" == true ]]; then
+        VERSION_TYPE="prerelease-only"
+    fi
+}
+
+# Find all go.mod files, excluding specified directories
+find_go_modules() {
+    local modules=()
+    local seen_modules=()
+
+    # Find all go.mod files
+    while IFS= read -r -d '' file; do
+        local dir=$(dirname "$file")
+        local rel_dir=$(get_relative_path "$dir" "$REPO_ROOT")
+
+        # Normalize relative directory (remove ./ prefix)
+        if [[ "$rel_dir" == "./" ]]; then
+            rel_dir="."
+        fi
+        rel_dir="${rel_dir#./}"
+
+        # Skip if in cmd directory or subdirectories
+        if [[ "$rel_dir" =~ ^cmd/ ]] || [[ "$rel_dir" == "cmd" ]]; then
+            continue
+        fi
+
+        # Skip if in internal/tools directory
+        if [[ "$rel_dir" == "internal/tools" ]]; then
+            continue
+        fi
+
+        # Skip if in idls directory (submodule)
+        if [[ "$rel_dir" == "idls" ]]; then
+            continue
+        fi
+
+        # Check for duplicates
+        local found_duplicate=false
+        for seen in "${seen_modules[@]}"; do
+            if [[ "$seen" == "$dir" ]]; then
+                found_duplicate=true
+                break
+            fi
+        done
+
+        if [[ "$found_duplicate" == false ]]; then
+            modules+=("$dir")
+            seen_modules+=("$dir")
+        fi
+    done < <(find "$REPO_ROOT" -name "go.mod" -print0)
+
+    printf '%s\n' "${modules[@]}"
+}
+
+# Get the current global version from all modules
+get_current_global_version() {
+    local all_tags=()
+
+    # Collect all version tags from all modules (ignore prerelease format warnings)
+    while IFS= read -r tag; do
+        if [[ "$tag" =~ v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+            all_tags+=("$tag")
+        fi
+    done < <(git tag -l)
```

Comment:
nit / nonblocking: 

I think this could be `all_tags=$(git tag -l | grep -E 'v[0-9]+\.[0-9]+\.[0-9]+')`